### DISCUSSION TITLE
Connection issues using USRP x310 on srsRAN Project 5G

### DISCUSSION THREAD

**Discussion by oguztudo** [srsran/srsRAN_Project#62] [General Help]
Hello everybody,

I have trouble running the gNB without any errors/warnings and there are multiple things going on here. I have an USRP x310 with the newest LTS UHD version, a GPS Antenna connected and locked. Additionally I have a 10 MHz Signal going into the REF IN Port of the USRP. I attached my settings to this mail and a screenshot of the console output.

1) When i run the gNB with the test configuration i get Error: failed to read sensors. Did i attach something wrong here?

2) I get the Warning that there is an attempt to set the tick rate to 0, which is being skipped. At this point i cannot verify (more to that on 3) if this is causing an error but a warning always gives a bad feeling. There can be rare occurances of Underflows (not more than once every 20 mins) but i do not get any Lates oder Overflows.

3) I tried different variations of the clock/sync variables and none of them work for me in a way that i can find the gNB with my Samsung Galaxy S23 Ultra. It uses a Snapdragon 8 Gen 2 as a processor utilizing a Qualcomm X70 modem. To check the operator list i go to Connections->Mobile Networks->Network Operators and use the search manually function. This step worked fine for me on finding the enb on srsRAN 4g.

I tried to give as much detail as possible but feel free to ask further if there are missing details for the understanding of what i am doing.
![testconf](https://user-images.githubusercontent.com/129198919/233355482-de473a41-b19b-4e48-8930-5d1efd8bf565.png)
[testconf.txt](https://github.com/srsran/srsRAN_Project/files/11285753/testconf.txt)

**Comment by robertfalkenberg** [comment]
Numerous COTS smartphones filter certain PLMNs according to internal blacklists/whitelists. This especially applies to 5G/NR, so the phone may see a 4G network but not a 5G-SA network with the same PLMN.

I'd suggest you try to configure your network with a different PLMN than 00101 (test network). For example 99970 (private network), 460xx (China), 262xx (Germany). The IMSI of your SIM doesn't have to match - you can use roaming.

**Reply by oguztudo** [reply]
Thank you for the quick reply. I tried out various PLMN combinations while paying attention on using a combination that supports 5G. Unfortunately this didn't do the trick. I even switched to a VIVO 5G device to make further tests which didn't help either.
I think my settings could not be too faulty, but I might as well be wrong on that one. They have been added right under the picture and are quite easy to miss to be honest.

Additional Info: I don't own a programmable USIM card. All the tests have been done using a Sim card from my operator.  Hopefully this is not the dealbreaker here...

**Reply by Moussa-Guemdani** [reply]
Hello @oguztudo ,
I would like to know what type of GPS you did use, as well as the cables between the GPS and the usrp x310?

Thank you.

**Comment by asheeshtripathi** [comment]
@oguztudo, I see Tx gain and Rx gain in your cell is set to 0. Please try with Tx gain 20 or 25 and Rx gain 25, I was able to scan 5G network with these gain values and as @robertfalkenberg mentioned using PLMN 00101 or 99970. I am able to see the network and also PRACH with PLMN 00101, 99970 and USRP x310 (also B210 with different Tx and Rx power). In my case, I used band n78 and n41 with subcarrier spacing 30 KHz. you can try with these settings as  Samsung Galaxy S23 Ultra support these bands. Once UE is able to scan the network and you don't see PRACH, you can try some with time_alignment_calibration. Here is my x310 gNB config file where attach worked. 
[gnb_rf_x310_tdd_n78_20mhz.txt](https://github.com/srsran/srsRAN_Project/files/11286970/gnb_rf_x310_tdd_n78_20mhz.txt)

**Reply by asheeshtripathi** [reply]
I was not using gpsdo, I used internal clocks.

**Reply by oguztudo** [reply]
Hello Asheesh, Thank you for your reply!
I tried various different gain values besides the one in the config file I added but i understand that a single file can suggest the usage of a single value. Coming from there I saw that you made some additional changes. Although being familiar with the expressions you have used for the qos block i wasn't aware that you can add those since they're not specified in the documentation of srsRAN Project.
Nonetheless i started the gnb with the file you added (commented out the time alignment calibration) and had no success yet again. I can verify that the USRP is transmitting at the frequency that is being set but i am not able to look furhter into detail until Monday...

**Reply by asheeshtripathi** [reply]
please follow discussion : [#28 ](https://github.com/srsran/srsRAN_Project/discussions/28)

**Reply by oguztudo** [reply]
Well it is not exactly the same issue since the network can not be seen by my COTS UE. I just know that the USRP is doing something in the set frequency but i can not say what it is exactly... at least at this stage.

**Comment by CBockelmann** [comment]
At least with my S21 it does not consistently show the 5G-SA networks. With my mobile operator sim I can see the network, but cannot connect. With a sysmocom simcard, however, I cannot see the network.

Try this:

1.	Install Google Telephone App from Play Store
2.	Enter in Google Telephone App: *#*#4636#*#*
3.	Select Phone Information
4.	Set Preferred Network Type: NR only

That overwrites the network settings in the samsung settings app to NR only. Now, the network should be visible.

**Reply by oguztudo** [reply]
Hello and thanks for the reply!
I am surprised that my answer was not posted, hence the late reply.
I tried that but my phone is not able to start any connection when i use that setting...

**Reply by CBockelmann** [reply]
You wrote, that you do not have a programmed SIM-card for the network, correct? In that case you should not be able to connect.

Do you see the network with the NR only setting? My phone behaved differently with different SIM cards, but changing to NR only at least consistently shows the srsRAN 5G network in the list if I search manually.

**Reply by oguztudo** [reply]
That is correct, i explicitly specified that because I was aware that connection with a non-programmable SIM-card is not possible. The main issue here is that i can not find the gNB (still) and utilizing a spectrum analyzer i can see... nothing! The spectrum does not change at all when i start the gNB and a connection with another USRP as a UE is not possible. My concern is that the gNB part of srsRAN does not communicate correctly with the USRP and I am currently trying to figure out where this issue comes from. 
The initial guess is that this comes from using a loopback address but I am not able to test this because my open5gs AMF does not start properly when changing the NGAP address of the AMF. So that's basically where I'm stuck right now until i can properly open a socket for the AMF.

**Reply by pgawlowicz** [reply]
Hi @oguztudo, any update on this issue?
